lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -592,6 +593,9 @@ sub failed_modules_with_needles {
             for my $needle (@{$detail->{needles}}) {
                 push( @needles, [ $needle->{name}, $counter ] );
             }
+            if (!@{$detail->{needles}}) {
+                push(@needles, [ undef, $counter ] );
```

Comment:
I see now the push - but we're seriously inconsistent with a lot of perl habits. Also the $job->{id} vs $job->{'id'} is killing me :)